### PR TITLE
Fixes for oracle automated test failures

### PR DESCRIPTION
**Pull request by chetank-yb**

### CODE REVIEW

File: migtests/tests/oracle-tests/ora-hr-db-sanity-test/validate [hunk 1]
```diff
@@ -82,16 +82,6 @@ def migration_completed_checks(tgt):
...
- 
-	# The following constraint checks are handled below
-	# Checking the constraint : jhist_date_interval CHECK (end_date > start_date)
-	# Unique index checks on job history table
- 	# Validate the check constraint on employees table
-	for type_check, query_and_code in QUERIES_CHECK.items():
-		query, code = query_and_code['query'], query_and_code['code']
-		chk_err_returned = tgt.run_query_and_chk_error(query, code)
-		print(f"Checking {type_check} ..", code, {chk_err_returned} )
-		assert chk_err_returned == True
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-hr-db-sanity-test/validate:94]: Why have you removed these checcks?

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
                            d1 DATE,
                            d2 DATE,
                            d3 DATE,
                            tz1 TIMESTAMP WITH TIME ZONE,
                            tz2 TIMESTAMP WITH TIME ZONE);
+                       
+CREATE OR REPLACE FUNCTION TRUNC_DATE(d1 in VARCHAR,int_val in VARCHAR)
+RETURN DATE
+IS
+trunced_date DATE;
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:11]: `truncated_date`

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+BEGIN
+    SELECT TRUNC(TO_DATE(d1,'DD-MON-YY'), int_val) into trunced_date FROM DUAL;
+RETURN trunced_date;
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:14]: Fix the indentation.

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
+END;
+/
+
+CREATE OR REPLACE FUNCTION TRUNC_TIME_STAMP(d1 in VARCHAR,int_val in VARCHAR)
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:18]: Format ALL SQL statements using any online SQL formatter.

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql [hunk 5]
```diff
@@ -12,13 +32,11 @@ DECLARE
...
+    d1_date := TRUNC_DATE('02-MAR-15', 'mi');
+    d2_date := TRUNC_DATE('02-MAR-16', 'ww');
+    d3_date :=TRUNC_TIME_STAMP('02-MAR-18', 'mm');
+    tz1_tz :=TRUNC_TIME_STAMP('02-MAR-18', 'mm');
+    tz2_ts := TRUNC_TIME_STAMP('02-MAR-19', 'YEAR');
+    INSERT INTO TRUNC_TEST (d1,d2,d3,tz1,tz2) VALUES(d1_date,d2_date,d3_date,tz1_tz,tz2_ts);
 END;
 /
-
-CREATE OR REPLACE SYNONYM TEST_SYNONYM for TRUNC_TEST_1;
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:24]: Why is this removed? Was this added by you?

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate [hunk 6]
```diff
@@ -17,9 +17,8 @@ def migration_completed_checks(tgt):
 		print(f"table_name: {table_name}, row_count: {got_row_count[table_name]}")
 		assert row_count == got_row_count[table_name]
 	
-	view_list = tgt.get_objects_of_type("VIEW", "test_mysql_views")
-	print("view_list:", view_list)
-	assert len(view_list) == 1
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate:22]: Why is this removed?

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate [hunk 7]
```diff
@@ -17,9 +17,8 @@ def migration_completed_checks(tgt):
...
+	function_count = tgt.get_functions_count("public")
+	assert function_count == 2
```
Comment (amit-yb) [inline migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate:21]: If object count is relatively small, it is better to make the check a bit stricter by checking even for the names. Consider something like:

```
function_names = tgt.list_function_names(schema="public")
assert set(function_names) == set(["func_name_1", "func_name_2"])
```

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql (hunk 2 above)
Comment (chetank-yb) [reply migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:11]: fixed

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql (hunk 5 above)
Comment (chetank-yb) [reply migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:24]: Yes, I removed it as the test schema was changed by me. This test was for validated truncated date functionality & the presence of synonyms here was not a requirement.

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate (hunk 6 above)
Comment (chetank-yb) [reply migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate:22]: The test schema was changed keeping in mind the objective and hence the assertion too.

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate (hunk 7 above)
Comment (chetank-yb) [reply migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/validate:20]: Added the assertion now

File: migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql (hunk 5 above)
Comment (priyanshi-yb) [reply migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:24]: @chetank-yb `CREATE OR REPLACE SYNONYM TEST_SYNONYM for TRUNC_TEST_1;` I added this DDL for Synonym test. Let me know if you already have synonym test, as we made a fix for Synonym now they are exporting correctly without source_schema name and hence can imported in target db schema. Hence, can have test for it.

Comment (chetank-yb) [reply migtests/tests/oracle-tests/ora-trunc-tests/ora-date-trunc-test/trunc_date_schema_data.sql:24]: I will add a new test for the synonym test @priyanshi-yb . The schema for this test is completely different now

**Review by priyanshi-yb — APPROVED** [review]
